disassemblers/ddisasm/ddisasmBBorRef.py

```diff
@@ -0,0 +1,305 @@
+import logging
+import optparse
+import subprocess
+import time
+from enum import Enum
+from pathlib import Path
+from typing import Optional
+
+import blocks_pb2
+import refInf_pb2 
+import gtirb
+from gtirb_capstone.instructions import GtirbInstructionDecoder
+
+logging.basicConfig(level = logging.INFO)
+
+
+class DdisasmError(Exception):
+    """
+    Exception raised when ddisasm fails
+    """
+    pass
+
+
+def disassemble(binary: Path, output: Path):
+    """
+    Run ddisasm on the given binary
+    
+    :param binary: the path to a binary to run ddisasm on
+    :param output: the path to the output
+    """
+    try:
+        p = subprocess.run(
+            ["ddisasm", binary, "--ir", output, "-j", "1"], check=True
+        )
+    except subprocess.CalledProcessError as e:
+        raise DdisasmError(f"ddisasm failed on {binary}: {e}")
+
+
+def addBB(block, pbFunc):
```

Comment:
```suggestion
def addBB(block:gtirb.CodeBlock, pbFunc):
```
what is the type of pbFunc?

Reply:
Added blocks_pb2.Function